eng/fv/tool-setup.rst

```diff
@@ -0,0 +1,239 @@
+.. SPDX-License-Identifier: CC-BY-SA-4.0
+
+.. Copyright (C) 2022 Trinity College Dublin
+
+.. _FormalToolSetup:
+
+Formal Tools Setup
+==================
+
+A key requirement for methods and tools for RTEMS qualification was that they
+comply to a large degree with RTEMS community guidelines. We wanted to avoid
+having large complex tools with large library dependencies, and have tools and
+notations that would be easy to maintain.
+
+After reviewing a range of methods and tools, we settled on a technique based on
+the SPIN model checker :cite:`Holzmann:1997:SPIN`, with its modelling language
+called Promela (https://spinroot.com/). Promela is quite a low-level modelling
+language that makes it easy to get close to code level, and is specifically 
+targeted to modelling software. It is one of the most widely used 
+model-checkers, both in industry and education. 
+It uses linear-time temporal logic (:term:`LTL`) to express properties of interest.
+
+It is open-source (https://github.com/nimble-code/Spin), actively maintained,
+and very easy to install, needing only  a C compiler and the ``lex``/``yacc``
+utilities.
+
+
+Installing Tools
+----------------
+
+The test generation tools are found in ``formal/promela/src``, written for
+Python3. That directory contains ``requirements.txt`` that identifies the
+required Python libraries. A key requirement is the Coconut package, that
+defines a form of pattern-matching syntax for Python. Source files have
+extension ``.coco`` and are compiled into equivalent Python scripts.
+
+We use the Python virtual environment facility here, as seen elsewhere in the
+RTEMS ecosystem. To build the tools, enter ``formal/promela/src`` and issue the 
+commands:
+
+.. code:: shell
+
+  make env
+  . env/bin/activate
+  make py
+
+The test generation tools need to be used from within this Python virtual 
+environment. Use the ``deactivate` command to exit from it.
+
+Test generation is managed at the top level by the script ``testbuilder.py``
+located in the top-level of ``formal/promela/src``.
+To avoid using (long) absolute pathnames, it helps to define the following 
+alias: 
+
+.. code-block:: shell
+
+  alias tbuild='python3 /..../formal/promela/src/testbuilder.py'
+
+We will use this alias in what follows.
+
+Tool Configuration
+------------------
+
+**The process described here makes changes to `rtems` directly rather than
+using the `spec` folder in `rtems-central`. There are already hooks in the
+`spec` folder that create ``model-0.yml`` with the right name, but this does
+not seem to have propagated to the `rtems` submodule. This should probably be 
```

Comment:
remove this text. you may mention that there are scripts available to assist this within rtems-central.

Reply:
Ok - I guess this is a flag to @sebhub  -  we should really join the dots here.

But perhaps this is a phase two that looks at closer integration of the formal tests into CI of some sort.

Reply:
I suggest to fix these integration issues after the documentation. The documentation should state how it should look like in the near future using the stuff which is already available.